### PR TITLE
some code improvements as suggested by intellij idea's code inspection

### PR DESCRIPTION
**Pull request by Meduris**
this one is still kinda work in progress and is open for discussions on potential performance improvements

### CODE REVIEW

File: src/main/java/gtclassic/fluid/GTFluid.java [hunk 1]
```diff
@@ -17,7 +17,16 @@
...
 	protected static Material material = Material.WATER;
-	GTMaterial mat = null;
+	GTMaterial mat;
+	
+	private GTFluid(GTMaterial mat, String base, String suffix, boolean isGaseous) {
+		super(mat.getDisplayName().toLowerCase() + suffix, new ResourceLocation(GTMod.MODID, "fluids/"
+				+ base), new ResourceLocation(GTMod.MODID, "fluids/flowing"));
+		this.mat = mat;
+		this.temperature = 300;
+		this.mapColor = mat.getColor().hashCode();
```
Comment (Speiger) [inline src/main/java/gtclassic/fluid/GTFluid.java]: getRGB. not hashcode. Even if they return the same thing. Its like using a comparator for a Object2ObjectFunction.

File: src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java [hunk 2]
```diff
@@ -44,22 +45,18 @@ public static void init() {
 	 * Stuff below for actual amp recipes and JEI iterators
 	 **/
 	public static IRecipeModifier[] value(int amount) {
-		return new IRecipeModifier[] { ModifierType.RECIPE_LENGTH.create((amount / 1) - 100) };
+		return new IRecipeModifier[] { ModifierType.RECIPE_LENGTH.create(amount - 100) };
 	}
 
 	public static void addAmplifierToJei(IRecipeInput[] inputs, IRecipeModifier[] modifiers, ItemStack... outputs) {
 		List<IRecipeInput> inlist = new ArrayList<>();
 		List<ItemStack> outlist = new ArrayList<>();
```
Comment (Speiger) [inline src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java]: Can be reduced to ObjectArrayList.wrap(array) so way less code.

File: src/main/java/gtclassic/tile/GTTileChargeOMat.java [hunk 3]
```diff
@@ -231,7 +231,7 @@ public void tryCharge() {
 		for (int i = 0; i < 9; ++i) {
 			if (!this.inventory.get(i).isEmpty()) {
 				if (hasEnergy()) {
```
Comment (Speiger) [inline src/main/java/gtclassic/tile/GTTileChargeOMat.java]: Should only iterate over slots if it has energy @e99999. (So if out of energy its should stop the loop.)

File: src/main/java/gtclassic/tile/GTTileFacing.java [hunk 4]
```diff
@@ -22,9 +23,7 @@ public GTTileFacing(EnumFacing... validRotations) {
 		super();
 		setFacing(EnumFacing.NORTH);
 		this.validRotations = new LinkedHashSet<>();
-		for (EnumFacing rotation : validRotations) {
-			this.validRotations.add(rotation);
-		}
+		Collections.addAll(this.validRotations, validRotations);
```
Comment (Speiger) [inline src/main/java/gtclassic/tile/GTTileFacing.java]: Use RotationList instead. Its just a Integer and has a lot of helper functions.

File: src/main/java/gtclassic/fluid/GTFluid.java (hunk 1 above)
Comment (Meduris) [reply src/main/java/gtclassic/fluid/GTFluid.java]: added in #86

File: src/main/java/gtclassic/tile/GTTileChargeOMat.java (hunk 3 above)
Comment (e99999) [reply src/main/java/gtclassic/tile/GTTileChargeOMat.java]: I have the hasEnergy() called on 233 below it so full batteries would still get passed to the output reguardless of the internal energy status.

File: src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java [hunk 5]
```diff
@@ -49,14 +48,12 @@ public static void init() {
...
-		List<ItemStack> outlist = new ArrayList<>();
-		Collections.addAll(inlist, inputs);
+		List<IRecipeInput> inlist = ObjectArrayList.wrap(inputs);
+		List<ItemStack> outlist = ObjectArrayList.wrap(outputs);
 		NBTTagCompound mods = new NBTTagCompound();
 		for (IRecipeModifier modifier : modifiers) {
 			modifier.apply(mods);
 		}
-		Collections.addAll(outlist, outputs);
 		addAmplifierToJei(inlist, new MachineOutput(mods, outlist));
```
Comment (Speiger) [inline src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java]: make it direct xD

File: src/main/java/gtclassic/tile/GTTileFacing.java (hunk 4 above)
Comment (Meduris) [reply src/main/java/gtclassic/tile/GTTileFacing.java]: fixed in ec4dd3fc6fe61fa075c8f98b06a70e0b3471871c

File: src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java (hunk 5 above)
Comment (Meduris) [reply src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java]: fixed in 4b65ed77f80277727fce24f4501157398784c45a

File: src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java (hunk 2 above)
Comment (Meduris) [reply src/main/java/gtclassic/recipe/GTRecipeUUAmplifier.java]: fixed in ec4dd3fc6fe61fa075c8f98b06a70e0b3471871c and 4b65ed77f80277727fce24f4501157398784c45a

File: src/main/java/gtclassic/block/GTBlockMachine.java [hunk 6]
```diff
@@ -46,18 +46,10 @@
 
 	String name;
 	String texture;
-	int size = 0;
+	int size;
 
 	public GTBlockMachine(String name, LocaleComp comp) {
-		super(Material.IRON);
-		this.name = name;
```
Comment (e99999) [inline src/main/java/gtclassic/block/GTBlockMachine.java]: hey @Meduris can you put all the constructor fixes like this into the other PR? they seem to be in both atm

Comment (Meduris) [reply src/main/java/gtclassic/block/GTBlockMachine.java]: that just happened when creating a separate branch for this changes based on the one for the first pull request. Once the first pull request is merged, I'll rebase this branch which should remove those commits

Comment (e99999) [reply src/main/java/gtclassic/block/GTBlockMachine.java]: 👌

**Pull request closed without merging**